pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -156,7 +156,7 @@ func serviceHostname(name, namespace, domainSuffix string) model.Hostname {
 
 // kubeToIstioServiceAccount converts a K8s service account to an Istio service account
 func kubeToIstioServiceAccount(saname string, ns string) string {
-	return spiffe.MustGenSpiffeURI(ns, saname)
+	return spiffe.GenSpiffeURI(ns, saname)
```

Comment:
Please make the kubeToIstioServiceAccount return error if saname is empty.

Reply:
I am hesitant to make this change so close to the 1.1 release. Right now, this causes no (reported) issues. If we make this fail, what will happen to the kube pods without a service account? From what I can tell, this will make these pods inaccessible. If we want to fail to generate a cert in this case that is one thing, but I think causing an error for all cases is too risky

Reply:
I agree that it's better to check in this PR after 1.1 release.
This will make these pods inaccessible, only if they rely on mTLS. I would like to further understand why the pods don't have service accounts.

Reply:
How about for 1.1 we just drop this log down to a warn or maybe debug, and after 1.1 release we make a proper fix?

Reply:
Sure. Please make the PR as small as possible if you want to check into 1.1.